main.nf

```diff
@@ -1,31 +1,34 @@
 // Ensure DSL2
 nextflow.enable.dsl = 2
 
+params.s3_file = "s3://iatlas-project-tower-bucket/s3_file.csv"
 //url for example CWL workflow
-params.cwl_file = "https://raw.githubusercontent.com/CRI-iAtlas/iatlas-workflows/develop/EPIC/workflow/steps/epic/epic.cwl"
-//example params file
-params.input_file = "${projectDir}/example_inputs/epic.json"
+params.cwl_file = "https://raw.githubusercontent.com/CRI-iAtlas/iatlas-workflows/develop/Immune_Subtype_Clustering/workflow/steps/immune_subtype_clustering/immune_subtype_clustering.cwl"
 
 //runs cwl workflow using url and params provided
 process EXECUTE_CWL_WORKFLOW {
+    debug true
     // containerOptions only work when run locally, aws batch volume mounting in nextflow.config for Tower runs
-    containerOptions = '-v "/var/run/docker.sock:/var/run/docker.sock" -v /tmp:/tmp'
+    containerOptions = '-v /var/run/docker.sock:/var/run/docker.sock -v /tmp:/tmp -v \$PWD:/input'
     container "ghcr.io/sage-bionetworks-workflows/nf-cwl-wrap:1.0"
 
     input:
     path cwl_file
-    path input_file
+    tuple path(input_file), path(data_file)
+
+    output:
+    path "*.tsv"
```

Comment:
Is there an efficient way to grab whatever is the output file from the CWL workflow and have that be the output? I'm assuming the output file wont always be `.tsv`, and this also emits the path to the input `.tsv` file in this case so not really ideal.

Reply:
Aside from `**`, which captures everything, your idea is a good one, but I think the output key of CWL files also differs based on what is specified in the CWL workflow. 

Some CWL workflows has the output as an input parameter, while others might not so it gets a bit tricky.  You could, parse through the dictionary of outputs and extract the paths, but I'm unsure of how that would translate to the CWL output binding.

Another option is to specify an output directory and just capture everything in that output directory (if thats allowed)

Reply:
What do you think about just using the `**` method for now, and leaving optimizing the output for later? Not sure what this will look like if incorporated into a larger process so maybe it would be better to hold off.